src/eterna/UndoBlock.ts

```diff
@@ -43,7 +43,7 @@ export interface TargetConditions {
     'custom-layout'?: ([number, number] | [null, null])[];
     'custom-reference'?: string;
     'custom-numbering'?: string;
-    annotations?: AnnotationDataBundle;
+    annotations?: AnnotationData[];
```

Comment:
Wanted to just check-in on the motivation behind your preference for using an array here. Definitely out of curiosity than criticism

Reply:
The TargetConditions interface is a field on the puzzle definition - it should only contain puzzle annotations, so there's no reason for it to hold an interface containing both puzzle and solution annotations.

Reply:
Got it!